*** gcc-4.0.2/gcc-4.0.2/gcc/config/pic30/pic30.c.original.1	2012-09-23 23:18:48.352666306 -0400
--- gcc-4.0.2/gcc-4.0.2/gcc/config/pic30/pic30.c	2012-09-23 23:19:42.454669939 -0400
***************
*** 10306,10317 ****
  /*
  ** Calculate the offset between two frame registers.
  */
! void pic30_initial_elimination_offset(int from, int to, int *poffset) {
    int fLeaf;
    int fInterrupt;
    int nLastReg;
    int regno;
!   int nOffset;
  
    fLeaf = current_function_is_leaf;
    fInterrupt = pic30_interrupt_function_p(current_function_decl);
--- 10306,10317 ----
  /*
  ** Calculate the offset between two frame registers.
  */
! void pic30_initial_elimination_offset(int from, int to, HOST_WIDE_INT *poffset) {
    int fLeaf;
    int fInterrupt;
    int nLastReg;
    int regno;
!   HOST_WIDE_INT nOffset;
  
    fLeaf = current_function_is_leaf;
    fInterrupt = pic30_interrupt_function_p(current_function_decl);
